internal/observer/opensearch/queries.go

```diff
@@ -582,3 +585,149 @@ func contains(slice []string, item string) bool {
 	}
 	return false
 }
+
+func (qb *QueryBuilder) BuildLogAlertingRuleQuery(params types.AlertingRuleRequest) map[string]interface{} {
+	filterConditions := []map[string]interface{}{
+		{
+			"range": map[string]interface{}{
+				"@timestamp": map[string]interface{}{
+					"gte":    "{{period_end}}||-" + params.Condition.Window,
+					"lte":    "{{period_end}}",
+					"format": "epoch_millis",
+				},
+			},
+		},
+		{
+			"term": map[string]interface{}{
+				labels.OSComponentID + ".keyword": params.Metadata.ComponentUID,
+			},
+		},
+		{
+			"term": map[string]interface{}{
+				labels.OSEnvironmentID + ".keyword": params.Metadata.EnvironmentUID,
+			},
+		},
+		{
+			"term": map[string]interface{}{
+				labels.OSProjectID + ".keyword": params.Metadata.ProjectUID,
+			},
+		},
+		{
+			"wildcard": map[string]interface{}{
+				"log": fmt.Sprintf("*%s*", params.Source.Query),
+			},
+		},
+	}
+
+	query := map[string]interface{}{
+		"size": 0,
+		"query": map[string]interface{}{
+			"bool": map[string]interface{}{
+				"filter": filterConditions,
+			},
+		},
+	}
+	return query
+}
+
+func (qb *QueryBuilder) BuildLogAlertingRuleMonitorBody(params types.AlertingRuleRequest) (map[string]interface{}, error) {
+	intervalDuration, err := time.ParseDuration(params.Condition.Interval)
+	if err != nil {
+		return nil, fmt.Errorf("invalid interval format: %w", err)
+	}
+	period := map[string]interface{}{
+		"interval": intervalDuration.Minutes(),
+		"unit":     "MINUTES",
+	}
+	monitorBody := map[string]interface{}{
+		"type":         "monitor",
+		"monitor_type": "query_level_monitor",
+		"name":         params.Metadata.Name,
+		"enabled":      params.Condition.Enabled,
+		"schedule": map[string]interface{}{
+			"period": period,
+		},
+		"inputs": []map[string]interface{}{
+			{
+				"search": map[string]interface{}{
+					"indices": []string{qb.indexPrefix + "*"},
+					"query":   qb.BuildLogAlertingRuleQuery(params),
+				},
+			},
+		},
+		"triggers": []map[string]interface{}{
+			{
+				"name":     "trigger-" + params.Metadata.Name,
+				"severity": "1",
+				"condition": map[string]interface{}{
+					"script": map[string]interface{}{
+						"source": "ctx.results[0].hits.total.value " + GetOperatorSymbol(params.Condition.Operator) + " " + strconv.FormatFloat(params.Condition.Threshold, 'f', -1, 64),
```

Comment:
When possible, let's use fmt.Sprintf to build the strings to improve the readability.